### DISCUSSION TITLE
Arabic text (RTL) rendering is not be displayed correctly in Terminal on Zed

### DISCUSSION THREAD

**Discussion by mjeragh** [zed-industries/zed#22458] [Feature Requests]
Title: Arabic text rendering issues in terminal

Description:
Currently experiencing issues with Arabic text rendering in Zed's terminal :

### Current Behavior
- Arabic characters appear disconnected
- Characters are not joining properly
- Right-to-left (RTL) text direction is not handled correctly
- Font ligatures for Arabic text are not working

### Expected Behavior
- Arabic characters should connect properly where appropriate
- Text should render right-to-left
- Character shapes should change based on their position in the word
- Similar to how macOS Terminal handles Arabic text rendering

### Environment
- Zed Version: [0.166.1]
- OS: macOS
- Font settings tried:
  - Arial Unicode MS
  - Microsoft Sans Serif
  - Menlo
  - System default fonts with Arabic support

### Example
When inserting Arabic text like "تسري احكام هذا القانون" into a MongoDB document:
- macOS Terminal: renders correctly with proper joining and RTL
- Zed: shows disconnected characters without proper joining

### Additional Notes
- The issue affects both the editor and integrated terminal
- System's native terminal (macOS Terminal.app) renders the same text correctly
- This affects Arabic text editing and database operations where Arabic content is involved

### Steps to Reproduce
1. Open Zed
2. Create a new Python file
3. Insert Arabic text in strings
4. Observe the disconnected characters and incorrect text rendering

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
